src/components/film_page/FilmPage.tsx

```diff
@@ -1,27 +1,80 @@
 import { useGetFilmByIdQuery } from "../../redux/slices/apiSlice";
-import { useAppSelector } from "../../redux/store/hooks";
+import noposter from "../../assets/noposter.jpg";
+import { SyntheticEvent } from "react";
+import { useParams } from "react-router";
+import { FilmIdObj } from "../../shared/types";
+import { useState } from "react";
+import { useAppSelector, useAppDispatch } from "../../redux/store/hooks";
+import { changeId } from "../../redux/slices/mainSlice";
+import { useEffect } from "react";
 
 const FilmPage = () => {
-  const id = useAppSelector((state) => state.mainSlice.id);
+  //fallback id
+  const filmIdFromStore = useAppSelector((store) => store.mainSlice.id);

```

Comment:
i don't see a point in keeping just a movie id in the store. We can always take it from the params
I would suggest following structure:
`const [movie, setMovie] = useState(null);`
 useEffect(() => {
  try to find the movie by id in the store, if found setMovie(foundMovie)
  else  we make an api request useGetFilmByIdQuery(id).then(setMovie)
  
  }, []);